flink-table/flink-table-planner/src/test/resources/org/apache/flink/table/planner/plan/batch/sql/join/SingleRowJoinTest.xml

```diff
@@ -237,20 +237,18 @@ LogicalProject(EXPR$0=[*($0, 0.1:DECIMAL(2, 1))])
     <Resource name="optimized exec plan">
       <![CDATA[
 Calc(select=[a2, EXPR$1])
-+- NestedLoopJoin(joinType=[InnerJoin], where=[(EXPR$1 > $f0)], select=[a2, EXPR$1, $f0], build=[right], singleRowJoin=[true])
-   :- Exchange(distribution=[any])
-   :  +- HashAggregate(isMerge=[true], groupBy=[a2], select=[a2, Final_SUM(sum$0) AS EXPR$1])
-   :     +- Exchange(distribution=[hash[a2]])
-   :        +- LocalHashAggregate(groupBy=[a2], select=[a2, Partial_SUM(a1) AS sum$0])
-   :           +- LegacyTableSourceScan(table=[[default_catalog, default_database, A, source: [TestTableSource(a1, a2)]]], fields=[a1, a2])(reuse_id=[1])
-   +- Exchange(distribution=[broadcast])
-      +- HashAggregate(isMerge=[false], select=[SINGLE_VALUE(EXPR$0) AS $f0])
-         +- Calc(select=[($f0 * 0.1) AS EXPR$0])
-            +- HashAggregate(isMerge=[true], select=[Final_SUM(sum$0) AS $f0])
-               +- Exchange(distribution=[single])
-                  +- LocalHashAggregate(select=[Partial_SUM(a1) AS sum$0])
-                     +- Calc(select=[a1])
-                        +- Reused(reference_id=[1])
++- MultipleInput(readOrder=[0,1], members=[\nNestedLoopJoin(joinType=[InnerJoin], where=[(EXPR$1 > $f0)], select=[a2, EXPR$1, $f0], build=[right], singleRowJoin=[true])\n:- HashAggregate(isMerge=[true], groupBy=[a2], select=[a2, Final_SUM(sum$0) AS EXPR$1])\n:  +- [#2] Exchange(distribution=[hash[a2]])\n+- [#1] Exchange(distribution=[broadcast])\n])
+   :- Exchange(distribution=[broadcast])
+   :  +- HashAggregate(isMerge=[false], select=[SINGLE_VALUE(EXPR$0) AS $f0])
+   :     +- Calc(select=[($f0 * 0.1) AS EXPR$0])
+   :        +- HashAggregate(isMerge=[true], select=[Final_SUM(sum$0) AS $f0])
+   :           +- Exchange(distribution=[single])
+   :              +- LocalHashAggregate(select=[Partial_SUM(a1) AS sum$0])
+   :                 +- Calc(select=[a1])
+   :                    +- LegacyTableSourceScan(table=[[default_catalog, default_database, A, source: [TestTableSource(a1, a2)]]], fields=[a1, a2])(reuse_id=[1])
+   +- Exchange(distribution=[hash[a2]])
+      +- LocalHashAggregate(groupBy=[a2], select=[a2, Partial_SUM(a1) AS sum$0])
+         +- Reused(reference_id=[1])
```

Comment:
Seems this plan becomes quite different. And I'm not sure yet whether it is expected.

Reply:
The `Exchange(distribution=[any])` is added in `DeadlockBreakupProcessor` if the shuffle mode is pipeline shuffle